sdcm/utils/version_utils.py

```diff
@@ -433,9 +433,9 @@ def get_branch_version_for_multiple_repositories(urls):
 def is_enterprise(scylla_version):
     """
     :param scylla_version: scylla version string
-    :return: True if this version string passed is a scylla enterprise version
+    :return: True if this version string passed is a scylla enterprise version (=hosted in scylladb/scylla-enterprise)
     """
-    return bool(re.search(r"^20[0-9]{2}.*", scylla_version))
+    return "6.2.3" < ComparableScyllaVersion(scylla_version) < "2025.1.0"
```

Comment:
This logic must not be changed.
A lot of stuff all over the SCT may become broken.

Change K8S related stuff only.

Reply:
why not? it's faulty logic. as in, prior to my changes,

`2025.1.10`
`2026.9.3`
`2029.0`

would all qualify as "enterprise" versions. Post my update, none of them will.

And what exactly is enterprise in the context of SCT anyway? The version which is present in the `scylladb/scylla-enterprise` docker hub repo?

Reply:
Before the move to the `source available` approach (`2025.1+`) for the ScyllaDB code base, this `is_enterprise` check was used as a switch for the set of features we expect to support by ScyllaDB clusters.

And whole SCT was dependent on the features it may work with, switch of the docker repo is just minor side effect.

And 2025.1+ has the same and more features than it was in the open source Scylla with versions like `6.2-`.

You don't want to update lots of other SCT places just to make `is_enterprise` be literally correct, need to revert back the changes to this `is_enterprise` function and literally update only K8S-related code.

Reply:
The upgrade logic is dependent on this logic, as Vallari pointed out we shouldn't change that (someone already did, and we needed to revert it)